be/src/storage/lake/compaction_policy.cpp

```diff
@@ -152,11 +153,14 @@ StatusOr<std::vector<RowsetPtr>> PrimaryCompactionPolicy::pick_rowsets(
     std::stringstream input_infos;
     while (!rowset_queue.empty()) {
         const auto& rowset_candidate = rowset_queue.top();
-        cur_compaction_result_bytes += rowset_candidate.stat.bytes;
+        cur_compaction_result_bytes += rowset_candidate.read_bytes();
         if (input_rowsets.size() > 0 && cur_compaction_result_bytes > config::update_compaction_result_bytes * 2) {
             break;
         }
         input_rowsets.emplace_back(std::make_shared<Rowset>(tablet, std::move(rowset_candidate.rowset_meta_ptr)));
+        if (has_dels != nullptr) {
+            has_dels->push_back(rowset_candidate.delete_bytes() > 0);
+        }
         input_infos << input_rowsets.back()->id() << "|";
 
         if (cur_compaction_result_bytes > config::update_compaction_result_bytes ||
```

Comment:
when all rowsets are bigger than config::update_compaction_result_bytes, it seems to stop compactions ?
config::update_compaction_result_bytes cannot dynamically adapt to variable tablet size？

Reply:
Good point!

Reply:
Done.